### PR TITLE
SYNCT-327:Enable disabling of Manual Sync to Parent and Manual Pull t…

### PR DESCRIPTION
**Pull request by reagan-meant**
issue: https://issues.openmrs.org/browse/SYNCT-327

Have made a check of the whitelist if it is disabled to prevent pulling from parent and pushing to child server...However I am failing with the unit tests to instatiate the parentfeedreader class

### CODE REVIEW

**Comment by reagan-meant** [comment]
Also on building the current code source of the sync2-module to get an omod file; it throws the following errors on uploading it to  a running openmrs-standalone instance with similar errors of failure to instatiate the parentfeedreader class..https://pastebin.com/WFkXL2JH

**Comment by reagan-meant** [comment]
@dkayiwa  I tried to add the dependency since the nullpointerexception was pointing to administrator class in the openmrs api

**Comment by dkayiwa** [comment]
@reagan-meant did the exception go away after adding that dependency?

**Comment by reagan-meant** [comment]
No, It doesnt..

**Comment by reagan-meant** [comment]
was borrowing this idea.. https://talk.openmrs.org/t/module-development-org-openmrs-api-apiexception-service-not-found/1780

**Comment by reagan-meant** [comment]
I just wanted to show you errors thrown despite adding it...

**Comment by dkayiwa** [comment]
Can you remove those that you have attempted but did not work?

File: api/src/test/java/org/openmrs/module/sync2/api/service/impl/atomfeed/LocalAtomfeedFeedReaderTest.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package org.openmrs.module.sync2.api.service.impl.atomfeed;
+
+
+public class LocalAtomfeedFeedReaderTest {
+	
+public void LocalAtomfeedFeedReader_shouldNotSyncIfPushIsDisabled() {
+
```
Comment (ODORA0) [inline api/src/test/java/org/openmrs/module/sync2/api/service/impl/atomfeed/LocalAtomfeedFeedReaderTest.java]: @reagan-meant What are we testing here?

Comment (reagan-meant) [reply api/src/test/java/org/openmrs/module/sync2/api/service/impl/atomfeed/LocalAtomfeedFeedReaderTest.java]: havent written tests for this yet..Still focusing on how to instatiate the parentfeedreader in the other test class...

**Comment by reagan-meant** [comment]
@dkayiwa I have tried using Mockito but somehow am still failing to instatiate it....Any help here? Thanks

**Comment by dkayiwa** [comment]
Can you ask on talk?

**Comment by tendomart** [comment]
@reagan-meant  also just a quick one , i know it's slightly painful process but writing tests before introducing a method is the best thing to do for  a new feature.

**Comment by reagan-meant** [comment]
@tendomart thanks...will consider that next time

**Comment by reagan-meant** [comment]
@dkayiwa  @mhawila I realised I was attempting an implementation that was already in place, however I have corrected to what the ticket required...checkout the visual changes in the jira attachments..Kind regards

**Comment by gracepotma** [comment]
@dkayiwa to review

**Pull request merged**